include/zephyr/dt-bindings/memory-attr/memory-attr-arm.h

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2023 Carlo Caione <ccaione@baylibre.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef ZEPHYR_INCLUDE_DT_BINDINGS_MEM_ATTR_ARM_H_
+#define ZEPHYR_INCLUDE_DT_BINDINGS_MEM_ATTR_ARM_H_
+
+#include <zephyr/sys/util_macro.h>
+#include <zephyr/dt-bindings/memory-attr/memory-attr.h>
+
+/*
+ * Architecture specific ARM MPU related attributes.
+ *
+ * This list is to seamlessy support the MPU regions configuration using DT and
+ * the `zephyr,memory-attr` property.
+ *
+ * This is legacy and it should NOT be extended further. If new MPU region
+ * types must be added, these must rely on the generic memory attributes.
+ */
+#define DT_MEM_ARM_MASK(x)		((x) & DT_MEM_ARCH_ATTR_MASK)
+#define DT_MEM_ARM(x)			((x) << DT_MEM_ARCH_ATTR_SHIFT)
+
+#define  ATTR_MPU_RAM			BIT(0)
+#define  ATTR_MPU_RAM_NOCACHE		BIT(1)
+#define  ATTR_MPU_FLASH			BIT(2)
+#define  ATTR_MPU_PPB			BIT(3)
+#define  ATTR_MPU_IO			BIT(4)
+#define  ATTR_MPU_EXTMEM		BIT(5)
+#define  ATTR_MPU_UNKNOWN		BIT(11) /* must be last */
```

Comment:
I wonder if all these `ATTR_<arch>_UNKNOWN` can be alias to something like `DT_MEM_ARCH_UNKNOWN`. If we ever need to change the allocation of bits, we don't have to change all of these `BIT(11)`.

Reply:
yes, makes sense. I'll make the change.